src/main/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuit.java

```diff
@@ -40,6 +40,23 @@
 @Slf4j
 @Component
 public class ReportMapperShortCircuit extends AbstractReportMapper {
+    private enum ConversionEquipmentType {
+        GENERATOR("generators", "generatorConversion", "disconnectedTerminalGenerator", "disconnectedTerminalGeneratorSummary"),
+        BATTERY("batteries", "batteryConversion", "disconnectedTerminalGenerator", "disconnectedTerminalBatterySummary");
+
+        public final String equipmentsLabel;
+        public final String conversionMessageKey;
+        public final String disconnectedTerminalMessageKey;
+        public final String summaryMessageKey;
+
+        ConversionEquipmentType(String equipmentsLabel, String conversionMessageKey, String disconnectedTerminalMessageKey, String summaryMessageKey) {
+            this.equipmentsLabel = equipmentsLabel;
+            this.conversionMessageKey = conversionMessageKey;
+            this.disconnectedTerminalMessageKey = disconnectedTerminalMessageKey;
+            this.summaryMessageKey = summaryMessageKey;
+        }
```

Comment:
You can replace that with `@AllArgsConstructor`.

Reply:
Done